support-workers/src/main/scala/com/gu/support/workers/lambdas/CreateZuoraSubscription.scala

```diff
@@ -333,7 +333,7 @@ object DigitalSubscriptionGiftRedemption {
     val zuoraUpdater = new ZuoraUpdater(zuoraService, catalogService)
     for {
       codeValidation <- codeValidator.getStatus(redemptionData.redemptionCode, Some(state.requestId.toString))
-      termDates <- codeValidation match {
+      (subscriptionNumber, termDates) <- codeValidation match {
```

Comment:
We need to know the subscriptionNumber for gift redemptions to create the data row so it's been added to `SendThankYouEmailDigitalSubscriptionGiftRedemptionState`